packages/utils/src/global.ts

```diff
@@ -26,23 +24,54 @@ interface SentryGlobal {
   };
 }
 
-const fallbackGlobalObject = {};
+// The code below for 'check' and 'GLOBAL' was copied from core-js before modification
+// https://github.com/zloirock/core-js/blob/1b944df55282cdc99c90db5f49eb0b6eda2cc0a3/packages/core-js/internals/global.js
+// core-js has the following licence:
+//
+// Copyright (c) 2014-2022 Denis Pushkarev
+//
+// Permission is hereby granted, free of charge, to any person obtaining a copy
+// of this software and associated documentation files (the "Software"), to deal
+// in the Software without restriction, including without limitation the rights
+// to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+// copies of the Software, and to permit persons to whom the Software is
+// furnished to do so, subject to the following conditions:
+//
+// The above copyright notice and this permission notice shall be included in
+// all copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+// THE SOFTWARE.
+
+/** Returns 'it' if it's the global object */
+function check(it: any): any {
+  // eslint-disable-next-line @typescript-eslint/no-unsafe-member-access
+  return it && it.Math == Math && it;
+}
+
+const GLOBAL =
+  check(typeof globalThis == 'object' && globalThis) ||
+  // eslint-disable-next-line no-restricted-globals
+  check(typeof window == 'object' && window) ||
+  check(typeof self == 'object' && self) ||
+  check(typeof global == 'object' && global) ||
+  (function (this: any) {
+    return this;
+  })() ||
+  {};
```

Comment:
```suggestion
const GLOBAL =
  (typeof globalThis == 'object' && check(globalThis)) ||
  // eslint-disable-next-line no-restricted-globals
  (typeof window == 'object' && check(window)) ||
  (typeof self == 'object' && check(self)) ||
  (typeof global == 'object' && check(global)) ||
  (function (this: any) {
    return this;
  })() ||
  {};
```

Similar to the above - I think it makes it clearer at a glance what's logic and what's value.